.github/workflows/npm-publish.yml

```diff
@@ -1,31 +1,74 @@
 name: NPM Publish
+
 on:
-  push:
+  workflow_run:
+    workflows: ["Node Build / Test / Lint"]
     branches: [main]
+    types:
+      - completed
```

Comment:
Could probably just do this on merges to main, any reason not to?

Reply:
I don't want this to run unless the linting and tests have already passed successfully.

Reply:
And I guess to clarify, someone technically _could_ push a commit straight to `main` without pushing a PR first (or at least I don't know if we're preventing that somehow, but didn't want to assume or depend on any git rules to enforce that).

Reply:
ah, gotcha. i'd say you can keep it as on `push` and add this validator on the `npm-publish.yml`:

```
 on:
   workflow_run:
     workflows: ["Node Build / Test / Lint"]
     types: [completed]

jobs:
     publish:
     if: ${{ github.event.workflow_run.conclusion == 'success'  && github.ref == 'refs/heads/main' }}
```

in theory this would trigger the publish workflow only if the node build/test/lint succeeded, then only run the publish job on the main branch.

alternatively you could consolidate to one workflow and add the publish step as an optional job that only runs on main branch + build/test success

Reply:
Yeah I had started down the path of adding it as an extra job to the `build-test-lint` workflow and decided against it but now I can't even remember what reason I had to abandon that.  I'll give it another go as it seems like that would be a simpler/easier to maintain approach.

Reply:
Ok I think that ends up looking better consolidated into one.  I'm going to merge and then see if it runs the `publish` job as it should now that I'll be pushing to `main`.